### PR TITLE
Re-introduce Retry logic for prepared statement caching 

### PR DESCRIPTION
**Pull request by AfsanehR**
- Added APIs for disableStatementPooling
- Set the disableStatementPooling to true by default
- Disable prepared statement caching by default 
- Introduce retry logic for missing handle exception. 

Fixes issue #610

### CODE REVIEW

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerPreparedStatement.java [hunk 1]
```diff
@@ -547,7 +547,7 @@ final void doExecutePreparedStatement(PrepStmtExecCmd command) throws SQLServerE
                 getNextResult();
             }
             catch (SQLException e) {
-                if (retryBasedOnFailedReuseOfCachedHandle(e, attempt))
+                if (retryBasedOnFailedReuseOfCachedHandle(e, attempt) && connection.isStatementPoolingEnabled())
```
Comment (TobiasSQL) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerPreparedStatement.java]: Suggest changing `retryBasedOnFailedReuseOfCachedHandle` to take care of the `connection.IsStatementPoolingEnabled()` checking, that is part of whether or not to reuse.

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerPreparedStatement.java [hunk 2]
```diff
@@ -565,13 +566,18 @@ else if (EXECUTE_UPDATE == executeMethod && null != resultSet) {
     
     /** Should the execution be retried because the re-used cached handle could not be re-used due to server side state changes? */
     private boolean retryBasedOnFailedReuseOfCachedHandle(SQLException e,
```
Comment (TobiasSQL) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerPreparedStatement.java]: Posting here as well as the method changed. Suggest changing retryBasedOnFailedReuseOfCachedHandle to take care of the connection.IsStatementPoolingEnabled() checking, that is part of whether or not to reuse.

Comment (v-afrafi) [reply src/main/java/com/microsoft/sqlserver/jdbc/SQLServerPreparedStatement.java]: done. Thanks!

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerPreparedStatement.java (hunk 1 above)
Comment (v-afrafi) [reply src/main/java/com/microsoft/sqlserver/jdbc/SQLServerPreparedStatement.java]: done.

**Pull request merged**